packages/nimble-components/src/table/index.ts

```diff
@@ -112,7 +106,7 @@ export class Table<
         const rows = this.table.getRowModel().rows;
         this.tableData = rows.map(row => {
             const rowState: TableRowState<TData> = {
-                record: row.original,
+                record: { ...row.original },
```

Comment:
hmmmmm this looks like it moves us from holding a reference to the original record object to make a copy. And it seems like this runs for all rows in the table. Is that the intention (i.e. so we are not holding onto references in the original object)?

If so, how often is refreshRows called? If it is more often than setData then maybe instead we should do a full copy once in setData instead and then here continue to just pass a reference since we own the full copy at that point?

Maybe we should just do that copy in setData instead so it's very clear where the cost of a copy is being made.

Also dates are unfortunately mutable objects, so for a proper clone we actually need to make a copy, ie `new Date(originalDate)`. 

I hadn't considered that before but this actually might be a useful change. The Date object instance really only has one piece of state associated with it, an number which is milliseconds since the unix epoch. Maybe the table should not take Date objects but instead take the number representation (ie `date.valueOf()` which is equal to `date.getTime()`);. If we make the user pass the number instead then we can make performant copies without concern like with the other primitive types. I don't think that's a huge amount of overhead for users either. Alternatively in our setData copy we could do that conversion for them.